<a href="https://colab.research.google.com/github/luisfelipebp/classification-depression/blob/main/data_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Verificando os tipos dos dados nulos:

MCAR: A ausência é totalmente aleatória e não tem relação com nenhum outro dado do conjunto de dados.

MAR: A ausência não é aleatória, mas pode ser explicada por outras variáveis presentes no conjunto de dados.

MNAR: A ausência depende do próprio valor que está faltando, como no caso de pessoas com salários muito altos que preferem não informar esse valor.

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Data-Science/Classificação/depression/me_cfs_vs_depression_dataset.csv')

In [ ]:
df

age  gender  sleep_quality_index  brain_fog_level  physical_pain_score  \
0     56    Male                  8.7              3.9                  9.2   
1     69    Male                  1.3              9.9                  4.2   
2     46  Female                  4.0              5.4                  4.8   
3     32  Female                  9.4              2.1                  2.9   
4     60  Female                  7.6              7.5                  6.4   
..   ...     ...                  ...              ...                  ...   
995   25  Female                  8.2              9.2                  8.8   
996   66    Male                  9.7              1.3                  2.4   
997   52    Male                  8.7              4.2                  8.7   
998   69    Male                  2.1              1.1                  9.9   
999   32    Male                  7.7              1.7                  7.5   

     stress_level  depression_phq9_score  fatigue_severity_scale_score  \
0             8.1                   10.0                           6.5   
1             9.9                   20.0                           7.0   
2             NaN                   24.0                           1.6   
3             3.8                   10.0                           6.8   
4             8.5                   17.0                           7.0   
..            ...                    ...                           ...   
995           3.6                   10.0                           6.9   
996           2.8                    9.0                           7.0   
997           3.4                    7.0                           7.0   
998           2.5                   15.0                           3.5   
999           1.9                   26.0                           5.4   

     pem_duration_hours  hours_of_sleep_per_night  pem_present  \
0                   9.0                       7.7            0   
1                  41.0                       8.4            1   
2                  13.0                       6.9            0   
3                  11.0                       7.5            0   
4                  46.0                       3.1            1   
..                  ...                       ...          ...   
995                29.0                       3.2            0   
996                34.0                       NaN            1   
997                40.0                       9.2            1   
998                37.0                       3.3            0   
999                21.0                       8.1            0   

           work_status social_activity_level exercise_frequency  \
0              Working                   Low              Daily   
1              Working                   Low              Often   
2    Partially working                   NaN             Rarely   
3          Not working                  High              Never   
4          Not working                   Low             Rarely   
..                 ...                   ...                ...   
995  Partially working                Medium                NaN   
996            Working             Very high             Rarely   
997            Working             Very high              Daily   
998  Partially working                   Low              Never   
999  Partially working              Very low             Rarely   

    meditation_or_mindfulness   diagnosis  
0                         Yes  Depression  
1                         Yes        Both  
2                         Yes  Depression  
3                         Yes  Depression  
4                          No        Both  
..                        ...         ...  
995                       Yes  Depression  
996                        No      ME/CFS  
997                       Yes      ME/CFS  
998                        No  Depression  
999                       Yes  Depression  

[1000 rows x 16 columns]

In [ ]:
df.isnull().sum()

age                              0
gender                           0
sleep_quality_index             47
brain_fog_level                 48
physical_pain_score             34
stress_level                    48
depression_phq9_score           22
fatigue_severity_scale_score    21
pem_duration_hours              24
hours_of_sleep_per_night        21
pem_present                      0
work_status                     47
social_activity_level           40
exercise_frequency              39
meditation_or_mindfulness       11
diagnosis                        0
dtype: int64

In [ ]:
percentual_nulos = (df.isnull().sum() / len(df)) * 100
print(percentual_nulos)

age                             0.0
gender                          0.0
sleep_quality_index             4.7
brain_fog_level                 4.8
physical_pain_score             3.4
stress_level                    4.8
depression_phq9_score           2.2
fatigue_severity_scale_score    2.1
pem_duration_hours              2.4
hours_of_sleep_per_night        2.1
pem_present                     0.0
work_status                     4.7
social_activity_level           4.0
exercise_frequency              3.9
meditation_or_mindfulness       1.1
diagnosis                       0.0
dtype: float64


In [ ]:
df_stress_nulo = df[df['stress_level'].isnull()]
df_stress_nao_nulo = df[df['stress_level'].notnull()]

In [ ]:
print("Descrição de 'age' para Stress NULO:")
print(df_stress_nulo['age'].describe())
print("Descrição de 'age' para Stress NÃO NULO:")
print(df_stress_nao_nulo['age'].describe())

Descrição de 'age' para Stress NULO:
count    48.000000
mean     47.666667
std      13.971602
min      23.000000
25%      34.750000
50%      50.000000
75%      60.500000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para Stress NÃO NULO:
count    952.000000
mean      44.219538
std       15.374782
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para Stress NULO:")
print(df_stress_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para Stress NÃO NULO:")
print(df_stress_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para Stress NULO:
gender
Male      0.645833
Female    0.354167
Name: proportion, dtype: float64
Contagem de 'gender' para Stress NÃO NULO:
gender
Male      0.517857
Female    0.482143
Name: proportion, dtype: float64


Age:
A idade média de quem não preencheu o estresse é de 47.7 anos.

A idade média de quem preencheu o estresse é de 44.2 anos.

A mesma tendência é vista na mediana (o valor do meio): 50 anos para o grupo nulo contra 45 para o grupo não nulo.

Conclusão Parcial: Pessoas um pouco mais velhas têm uma probabilidade maior de deixar o campo stress_level em branco.

Gender:
No grupo que não preencheu o estresse, 64.6% são do gênero masculino (Male).

No grupo que preencheu, a proporção de homens é de 51.8%.

Conclusão Parcial: A diferença aqui é ainda mais gritante. Homens neste dataset têm uma tendência significativamente maior a não responder sobre seu nível de estresse em comparação com as mulheres.

In [ ]:
df_brain_fog_level_nulo = df[df['brain_fog_level'].isnull()]
df_brain_fog_level_nao_nulo = df[df['brain_fog_level'].notnull()]

In [ ]:
print("Descrição de 'age' para Brain fog level NULO:")
print(df_brain_fog_level_nulo['age'].describe())
print("Descrição de 'age' para Brain fog level Não NULO:")
print(df_brain_fog_level_nao_nulo['age'].describe())

Descrição de 'age' para Brain fog level NULO:
count    48.000000
mean     47.520833
std      15.586396
min      18.000000
25%      39.500000
50%      49.000000
75%      61.250000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para Brain fog level Não NULO:
count    952.000000
mean      44.226891
std       15.299498
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para Brain fog level NULO:")
print(df_brain_fog_level_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para Brain fog level Não NULO:")
print(df_brain_fog_level_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para Brain fog level NULO:
gender
Male      0.645833
Female    0.354167
Name: proportion, dtype: float64
Contagem de 'gender' para Brain fog level Não NULO:
gender
Male      0.517857
Female    0.482143
Name: proportion, dtype: float64


Os mesmos resultados que o stress level para brain fog level


In [ ]:
df_physical_pain_score_nulo = df[df['physical_pain_score'].isnull()]
df_physical_pain_score_nao_nulo = df[df['physical_pain_score'].notnull()]

In [ ]:
print("Descrição de 'age' para Phyisical Pain NULO:")
print(df_physical_pain_score_nulo['age'].describe())
print("Descrição de 'age' para  Phyisical Pain Não NULO:")
print(df_physical_pain_score_nao_nulo['age'].describe())

Descrição de 'age' para Phyisical Pain NULO:
count    34.000000
mean     40.794118
std      13.644356
min      19.000000
25%      31.500000
50%      40.000000
75%      48.250000
max      69.000000
Name: age, dtype: float64
Descrição de 'age' para  Phyisical Pain Não NULO:
count    966.000000
mean      44.511387
std       15.368408
min       18.000000
25%       32.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para  Phyisical Pain NULO:")
print(df_physical_pain_score_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para  Phyisical Pain Não NULO:")
print(df_physical_pain_score_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para  Phyisical Pain NULO:
gender
Female    0.529412
Male      0.470588
Name: proportion, dtype: float64
Contagem de 'gender' para  Phyisical Pain Não NULO:
gender
Male      0.52588
Female    0.47412
Name: proportion, dtype: float64


In [ ]:
df_sleep_quality_index_nulo = df[df['sleep_quality_index'].isnull()]
df_sleep_quality_index_nao_nulo = df[df['sleep_quality_index'].notnull()]

In [ ]:
print("Descrição de 'age' para sleep_quality_index NULO:")
print(df_sleep_quality_index_nulo['age'].describe())
print("Descrição de 'age' para sleep_quality_index NÃO NULO:")
print(df_sleep_quality_index_nao_nulo['age'].describe())

Descrição de 'age' para sleep_quality_index NULO:
count    47.000000
mean     44.765957
std      17.319520
min      18.000000
25%      28.500000
50%      46.000000
75%      62.000000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para sleep_quality_index NÃO NULO:
count    953.000000
mean      44.366212
std       15.226345
min       18.000000
25%       32.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'stress_level' para sleep_quality_index NULO:")
print(df_sleep_quality_index_nulo['stress_level'].describe())
print("Contagem de 'stress_level' para sleep_quality_index NÃO NULO:")
print(df_sleep_quality_index_nao_nulo['stress_level'].describe())

Contagem de 'stress_level' para sleep_quality_index NULO:
count    45.000000
mean      5.535556
std       2.816522
min       1.100000
25%       3.000000
50%       5.600000
75%       7.800000
max      10.000000
Name: brain_fog_level, dtype: float64
Contagem de 'stress_level' para sleep_quality_index NÃO NULO:
count    907.000000
mean       5.615546
std        2.642335
min        1.000000
25%        3.300000
50%        5.800000
75%        7.900000
max       10.000000
Name: brain_fog_level, dtype: float64


In [ ]:
df_depression_phq9_score_nulo = df[df['depression_phq9_score'].isnull()]
df_depression_phq9_score_nao_nulo = df[df['depression_phq9_score'].notnull()]

In [ ]:
print("Descrição de 'age' para depression_phq9_score NULO:")
print(df_depression_phq9_score_nulo['age'].describe())
print("Descrição de 'age' para depression_phq9_score NÃO NULO:")
print(df_depression_phq9_score_nao_nulo['age'].describe())

Descrição de 'age' para depression_phq9_score NULO:
count    22.000000
mean     47.727273
std      13.980196
min      18.000000
25%      42.250000
50%      47.500000
75%      57.750000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para depression_phq9_score NÃO NULO:
count    978.000000
mean      44.309816
std       15.348664
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para depression_phq9_score NULO:")
print(df_depression_phq9_score_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para depression_phq9_score NÃO NULO:")
print(df_depression_phq9_score_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para depression_phq9_score NULO:
gender
Male      0.545455
Female    0.454545
Name: proportion, dtype: float64
Contagem de 'gender' para depression_phq9_score NÃO NULO:
gender
Male      0.523517
Female    0.476483
Name: proportion, dtype: float64


In [ ]:
df_fatigue_severity_scale_score_nulo = df[df['fatigue_severity_scale_score'].isnull()]
df_fatigue_severity_scale_score_nao_nulo = df[df['fatigue_severity_scale_score'].notnull()]

In [ ]:
print("Descrição de 'age' para fatigue_severity_scale_score NULO:")
print(df_fatigue_severity_scale_score_nulo['age'].describe())
print("Descrição de 'age' para fatigue_severity_scale_score NÃO NULO:")
print(df_fatigue_severity_scale_score_nao_nulo['age'].describe())

Descrição de 'age' para fatigue_severity_scale_score NULO:
count    21.000000
mean     38.047619
std      15.044854
min      19.000000
25%      28.000000
50%      38.000000
75%      42.000000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para fatigue_severity_scale_score NÃO NULO:
count    979.000000
mean      44.520940
std       15.306358
min       18.000000
25%       32.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para fatigue_severity_scale_score NULO:")
print(df_fatigue_severity_scale_score_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para fatigue_severity_scale_score NÃO NULO:")
print(df_fatigue_severity_scale_score_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para fatigue_severity_scale_score NULO:
gender
Male      0.571429
Female    0.428571
Name: proportion, dtype: float64
Contagem de 'gender' para fatigue_severity_scale_score NÃO NULO:
gender
Male      0.522983
Female    0.477017
Name: proportion, dtype: float64


In [ ]:
df_pem_duration_hours_nulo = df[df['pem_duration_hours'].isnull()]
df_pem_duration_hours_nao_nulo = df[df['pem_duration_hours'].notnull()]

In [ ]:
print("Descrição de 'age' para pem_duration_hours NULO:")
print(df_pem_duration_hours_nulo['age'].describe())
print("Descrição de 'age' para pem_duration_hours NÃO NULO:")
print(df_pem_duration_hours_nao_nulo['age'].describe())

Descrição de 'age' para pem_duration_hours NULO:
count    24.00000
mean     46.12500
std      14.63561
min      19.00000
25%      33.75000
50%      45.00000
75%      56.75000
max      70.00000
Name: age, dtype: float64
Descrição de 'age' para pem_duration_hours NÃO NULO:
count    976.000000
mean      44.342213
std       15.342841
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para pem_duration_hours NULO:")
print(df_pem_duration_hours_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para pem_duration_hours NÃO NULO:")
print(df_pem_duration_hours_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para pem_duration_hours NULO:
gender
Male      0.625
Female    0.375
Name: proportion, dtype: float64
Contagem de 'gender' para pem_duration_hours NÃO NULO:
gender
Male      0.521516
Female    0.478484
Name: proportion, dtype: float64


In [ ]:
df_hours_of_sleep_per_night_nulo = df[df['hours_of_sleep_per_night'].isnull()]
df_hours_of_sleep_per_night_nao_nulo = df[df['hours_of_sleep_per_night'].notnull()]

In [ ]:
print("Descrição de 'age' para hours_of_sleep_per_night NULO:")
print(df_hours_of_sleep_per_night_nulo['age'].describe())
print("Descrição de 'age' para hours_of_sleep_per_night NÃO NULO:")
print(df_hours_of_sleep_per_night_nao_nulo['age'].describe())

Descrição de 'age' para hours_of_sleep_per_night NULO:
count    21.000000
mean     43.380952
std      17.633707
min      19.000000
25%      25.000000
50%      42.000000
75%      61.000000
max      67.000000
Name: age, dtype: float64
Descrição de 'age' para hours_of_sleep_per_night NÃO NULO:
count    979.000000
mean      44.406537
std       15.277869
min       18.000000
25%       32.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para hours_of_sleep_per_night NULO:")
print(df_hours_of_sleep_per_night_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para hours_of_sleep_per_night NÃO NULO:")
print(df_hours_of_sleep_per_night_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para hours_of_sleep_per_night NULO:
gender
Male      0.571429
Female    0.428571
Name: proportion, dtype: float64
Contagem de 'gender' para hours_of_sleep_per_night NÃO NULO:
gender
Male      0.522983
Female    0.477017
Name: proportion, dtype: float64


In [ ]:
df_work_status_nulo = df[df['work_status'].isnull()]
df_work_status_nao_nulo = df[df['work_status'].notnull()]

In [ ]:
print("Descrição de 'age' para work_status NULO:")
print(df_work_status_nulo['age'].describe())
print("Descrição de 'age' para work_status NÃO NULO:")
print(df_work_status_nao_nulo['age'].describe())

Descrição de 'age' para work_status NULO:
count    47.000000
mean     48.297872
std      15.128313
min      23.000000
25%      38.500000
50%      50.000000
75%      61.000000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para work_status NÃO NULO:
count    953.000000
mean      44.192025
std       15.313114
min       18.000000
25%       31.000000
50%       45.000000
75%       56.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para work_status NULO:")
print(df_work_status_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para work_status NÃO NULO:")
print(df_work_status_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para work_status NULO:
gender
Male      0.595745
Female    0.404255
Name: proportion, dtype: float64
Contagem de 'gender' para work_status NÃO NULO:
gender
Male      0.520462
Female    0.479538
Name: proportion, dtype: float64


In [ ]:
df_social_activity_level_nulo = df[df['social_activity_level'].isnull()]
df_social_activity_level_nao_nulo = df[df['social_activity_level'].notnull()]

In [ ]:
print("Descrição de 'age' para social_activity_level NULO:")
print(df_social_activity_level_nulo['age'].describe())
print("Descrição de 'age' para social_activity_level NÃO NULO:")
print(df_social_activity_level_nao_nulo['age'].describe())

Descrição de 'age' para social_activity_level NULO:
count    40.000000
mean     46.400000
std      15.885488
min      20.000000
25%      35.000000
50%      46.500000
75%      58.250000
max      70.000000
Name: age, dtype: float64
Descrição de 'age' para social_activity_level NÃO NULO:
count    960.000000
mean      44.301042
std       15.300535
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para social_activity_level NULO:")
print(df_social_activity_level_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para social_activity_level NÃO NULO:")
print(df_social_activity_level_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para social_activity_level NULO:
gender
Male      0.625
Female    0.375
Name: proportion, dtype: float64
Contagem de 'gender' para social_activity_level NÃO NULO:
gender
Male      0.519792
Female    0.480208
Name: proportion, dtype: float64


In [ ]:
df_exercise_frequency_nulo = df[df['exercise_frequency'].isnull()]
df_exercise_frequency_nao_nulo = df[df['exercise_frequency'].notnull()]

In [ ]:
print("Descrição de 'age' para exercise_frequency NULO:")
print(df_exercise_frequency_nulo['age'].describe())
print("Descrição de 'age' para exercise_frequency NÃO NULO:")
print(df_exercise_frequency_nao_nulo['age'].describe())

Descrição de 'age' para exercise_frequency NULO:
count    39.000000
mean     40.923077
std      14.369897
min      19.000000
25%      27.000000
50%      42.000000
75%      51.000000
max      67.000000
Name: age, dtype: float64
Descrição de 'age' para exercise_frequency NÃO NULO:
count    961.000000
mean      44.525494
std       15.349594
min       18.000000
25%       32.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para exercise_frequency NULO:")
print(df_exercise_frequency_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para exercise_frequency NÃO NULO:")
print(df_exercise_frequency_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para exercise_frequency NULO:
gender
Female    0.512821
Male      0.487179
Name: proportion, dtype: float64
Contagem de 'gender' para exercise_frequency NÃO NULO:
gender
Male      0.525494
Female    0.474506
Name: proportion, dtype: float64


In [ ]:
df_meditation_or_mindfulness_nulo = df[df['meditation_or_mindfulness'].isnull()]
df_meditation_or_mindfulness_nao_nulo = df[df['meditation_or_mindfulness'].notnull()]

In [ ]:
print("Descrição de 'age' para meditation_or_mindfulness NULO:")
print(df_meditation_or_mindfulness_nulo['age'].describe())
print("Descrição de 'age' para meditation_or_mindfulness NÃO NULO:")
print(df_meditation_or_mindfulness_nao_nulo['age'].describe())

Descrição de 'age' para meditation_or_mindfulness NULO:
count    11.000000
mean     40.000000
std      13.114877
min      18.000000
25%      32.500000
50%      44.000000
75%      50.000000
max      57.000000
Name: age, dtype: float64
Descrição de 'age' para meditation_or_mindfulness NÃO NULO:
count    989.000000
mean      44.433771
std       15.343071
min       18.000000
25%       31.000000
50%       45.000000
75%       57.000000
max       70.000000
Name: age, dtype: float64


In [ ]:
print("Contagem de 'gender' para meditation_or_mindfulness NULO:")
print(df_meditation_or_mindfulness_nulo['gender'].value_counts(normalize=True))
print("Contagem de 'gender' para meditation_or_mindfulness NÃO NULO:")
print(df_meditation_or_mindfulness_nao_nulo['gender'].value_counts(normalize=True))

Contagem de 'gender' para meditation_or_mindfulness NULO:
gender
Female    0.545455
Male      0.454545
Name: proportion, dtype: float64
Contagem de 'gender' para meditation_or_mindfulness NÃO NULO:
gender
Male      0.524772
Female    0.475228
Name: proportion, dtype: float64
